### ISSUE TITLE
Error message when no value for the selected analyte

### ISSUE THREAD

**Issue by Fasjul** [tuneinsight/glowing-bear-medco#46]
The first error message stating no patient match the criteria, is good. The second error message warning about an error 500 is not useful to the user and can be misleading (can be interpreted as if the system is malfunctioning)

Also, is it possible to Incorporating an early "break point", when no entries to a query are returned?
It seems that the statistical calculations (even with no values) are still taking place, which increases the users waiting time.

**Comment by f-marino** [comment]
Can you give an example on how to reproduce the issue?

**Comment by f-marino** [comment]
Then not sure what you mean by early break point. Is it in the aggregation phase, or in the statistical computations that happen in the browser after aggregation.

- In the first case, even if the local query returned 0 patients, we still have to run the aggregation across the nodes, as the other ones may have returned some. Unless we want to explicitly enforce the fact that all nodes must contribute some data.
- In the second case, is the statistical library actually doing something when no observations are returned?

**Comment by Fasjul** [comment]
error messages are triggered on the following query:

Criterion : 
/sphn/sphn:SPHNConcept/sphn:FOPHDiagnosis/sphn:hasFOPHDiagnosisCode/icd-10-gm:ICD-10-GM/icd-10-gm:01/icd-10-gm:A15-A19/icd-10-gm:A15.-/icd-10-gm:A15.0
Analyte: Biosample/has subjectAge

**Comment by Fasjul** [comment]
About the breakpoint, I am unsure (not my code) if the explore query returns aggregated results before the statistics query is triggered (in which case we could stop the whole process), or if only one aggregation is done at the very end of both queries

**Comment by f-marino** [comment]
> About the breakpoint, I am unsure (not my code) if the explore query returns aggregated results before the statistics query is triggered (in which case we could stop the whole process), or if only one aggregation is done at the very end of both queries

Nope, the result of the explore query is not aggregated, is used as constraint to the statistics query, which is then aggregated.

**Comment by f-marino** [comment]
> error messages are triggered on the following query:
> 
> Criterion :
> /sphn/sphn:SPHNConcept/sphn:FOPHDiagnosis/sphn:hasFOPHDiagnosisCode/icd-10-gm:ICD-10-GM/icd-10-gm:01/icd-10-gm:A15-A19/icd-10-gm:A15.-/icd-10-gm:A15.0
> Analyte: Biosample/has subjectAge

This is what I get. I guess we can consider this fixed?

<img width="952" alt="image" src="https://user-images.githubusercontent.com/7016414/190388714-6fe97d92-936c-4bb9-b5de-28d9ded1e7c1.png">

**Comment by Fasjul** [comment]
Yes. It's not working this way on the current deployed version for bioref though. Is your screenshot from unpublished developments?

**Comment by f-marino** [comment]
yes, you will have it in the next release
